### ISSUE TITLE
On macOS, doc window's red "Close" button shows dot even after saved

### ISSUE THREAD

**Issue by dtgriscom** [typora/typora-issues#3596]
Usually, the red "Close" button at the top left of every window shows a dot if the file hasn't been saved. Sometimes, however, saving a file in Typora does NOT make the dot go away. The window can still be closed without a "Discard changes?" warning.

- macOS 10.13.6
- Typora 0.9.9.33.2 (4442)

One perhaps-relevant note: this is on a networked volume, mounted from a Linux box using sshfs. I haven't tried this on locally-saved files, so I don't know if that makes a difference.

**Comment by abnerlee** [comment]
relates #3981

Could you try the system built-in TextEditor.app? If it is the same behavior, then it is behavior of macOS.
Also, could you check if the content is saved, even no "Discard changes" shows up.

**Comment by dtgriscom** [comment]
With Typora on internal drive, this behavior doesn't occur. With Typora on networked (sshfs) drive, this **sometimes** happens. However, I just found a related strange behavior on a network drive that seems reliable:

1. Open file in Typora
2. Type some text; note that dot appears in "Close" button
3. Save: note that the dot (generally and correctly) disappears
4. Press Command-Z to undo, so that you've undone the last change and the screen no longer matches the saved file. Here, the dot does NOT reappear; it should.

Now, I tried this with the built-in TextEditor application. To my surprise, I get the same behavior. So, it's something in macOS.
